src/topology/homeomorph.lean

```diff
@@ -134,6 +134,17 @@ h.quotient_map.2.symm
 protected lemma embedding (h : α ≃ₜ β) : embedding h :=
 ⟨h.inducing, h.injective⟩
 
+/-- Homeomorphism given an embedding. -/
+noncomputable def of_embedding (f : α → β) (hf : embedding f) : α ≃ₜ (set.range f) :=
+{ continuous_to_fun := continuous_subtype_mk _ hf.continuous,
+  continuous_inv_fun := begin
+    rw hf.continuous_iff,
+    convert continuous_subtype_coe using 1,
+    ext b,
+    simpa [subtype.ext_iff] using (equiv.of_injective f hf.inj).right_inv b,
```

Comment:
This kind of proof very often hints there is a missing simp lemma somewhere. I recommend adding next to the definition of `equiv.of_injective`:
```lean
@[simp]
lemma equiv.self_comp_of_injective_symm (f : α → β) (hf : injective f) : f ∘ ((equiv.of_injective f hf).symm) = coe :=
funext (λ x, equiv.apply_of_injective_symm f hf x)
```
and replace this proof by:
```lean
continuous_inv_fun := by simp [hf.continuous_iff, continuous_subtype_coe],
```